llvm/include/llvm/Analysis/OptimizationRemarkEmitter.h

```diff
@@ -69,6 +69,11 @@ class OptimizationRemarkEmitter {
   /// Output the remark via the diagnostic handler and to the
   /// optimization record file.
   void emit(DiagnosticInfoOptimizationBase &OptDiag);
+  /// Also allow r-value for OptDiag to allow emitting a temporarily-constructed
+  /// diagnostic.
+  void emit(DiagnosticInfoOptimizationBase &&OptDiag) {
+    emit(static_cast<DiagnosticInfoOptimizationBase &>(OptDiag));
```

Comment:
Is the cast needed here? (eg: https://godbolt.org/z/rT8xnzfn3 )

Reply:
No, it's not once we have an r-value. Removed.